File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
N=int(input())
time = [[0]*2 for _ in range(N)]
for i in range(N):
    s, e = map(int, input().split())
    time[i][0] = s
    time[i][1] = e

time.sort(key = lambda x: (x[1], x[0]))

count=1
end=time[0][1]
for i in range(1,N):
    if time[i][0]>=end:
        count+=1
        end=time[i][1]
print(count)

File: 백준/Silver/1931. 회의실 배정/회의실 배정.java
import java.io.*;
import java.util.*;

public class Main {
	static int N;
	static Time[] times;
	static class Time implements Comparable<Time>{
		int start;
		int end;
		
		public Time(int start, int end) {
			this.start = start;
			this.end = end;
		}

		@Override
		public int compareTo(Time o) {
			return this.end!=o.end ? 
					Integer.compare(this.end, o.end):
						Integer.compare(this.start,o.start);
		}
	}
	static StringBuilder sb = new StringBuilder();
	public static void main(String[] args) throws Exception{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		N = Integer.parseInt(br.readLine());
		times = new Time[N];
		for(int i=0; i<N; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			times[i]=new Time(Integer.parseInt(st.nextToken()), Integer.parseInt(st.nextToken()));
		}
		Arrays.sort(times);
		List<Time> list = new ArrayList<>();
		list.add(times[0]); //첫회의 선택
		for(int j=1; j<N; j++) {
			if(list.get(list.size()-1).end <= times[j].start) {
				list.add(times[j]);
			}
		}
		
		sb.append(list.size());
		
		System.out.println(sb);
		br.close();
	}
}